js/calendar.js

```diff
@@ -83,3 +91,153 @@ function refreshCalendar(nextState) {
+		var endMinute = ("0" + eventData.end.getMinutes()).slice(-2);
+		var startHour = ("0" + eventData.start.getHours()).slice(-2);
+		var endHour = ("0" + eventData.end.getHours()).slice(-2);
+		var startDay = eventData.start.getDate();
+		var endDay = eventData.end.getDate();
+		var startMonth = eventData.start.getMonth();
+		var endMonth = eventData.end.getMonth();
+		var startYear = eventData.start.getFullYear()
+		var endYear = eventData.end.getFullYear();
+		const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+		var startDate = "";
+		var endDate = "";
+		if (startYear != endYear) {
+			// years differ - "dd MMM YYYY - dd MMM YYYY"
+			startDate = startDay + space + months[startMonth] + space + startYear;
+			endDate = endDay + space + months[endMonth] + space + endYear;
+			if (!allDay) {
+				// "dd MMM YYYY HH:MM - dd MMM YYYY HH:MM"
+				startDate += space + startHour + colon + startMinute;
+				endDate += space + startHour + colon + startMinute;
+			}
+		} else if (startMonth != endMonth) {
+			// same year, different month - "dd MMM - dd MMM"
+			startDate = startDay + space + months[startMonth];
+			endDate = endDay + space + months[endMonth];
+			if (!allDay) {
+				// "dd MMM HH:MM - dd MMM HH:MM"
+				startDate += space + startHour + colon + startMinute;
+				endDate += space + startHour + colon + startMinute;
+			}
+		} else if (startDay != endDay) {
+			// same year, same month, different day - "dd - dd MMM YYYY"
+			if (allDay) {
+				startDate = startDay;
+				endDate = endDay + space + months[endMonth] + space + endYear;
+			} else {
+				// "dd MMMM YYYY HH:MM - dd MMM YYYY HH:MM"
+				startDate = startDay + space + months[startMonth] + space + startHour + colon + endMinute;
+				endDate = endDay + space + months[endMonth] + space + endHour + colon + endMinute;
+			}
+		} else {
+			// same day - "dd MMM HH:MM - HH:MM"
+			startDate = startDay + space + months[startMonth] + space + startHour + colon + startMinute;
+			endDate = endHour + colon + endMinute;
+
+		}
+		date = startDate + dash + endDate;
+
+		//	location
+		location = eventDataExtended.location;
+
+		// description 
+		description = parseDescription(eventDataExtended.description);
+
+		// Facebook event button
+		let eventFacebookLinkElement = document.getElementById("facebook-link");
+		let link = parseFacebookLink(eventDataExtended.description);
+		if (link) {
+			eventFacebookLinkElement.onclick = () => {window.open(link)};
+			eventFacebookLinkElement.style.display = "inline";
+		} else {
+			// catches null regex matches
+			eventFacebookLinkElement.style.display = "none";
+		}
+	
+
+		// set text
+		document.getElementById("event-text-title").textContent = title;
+		document.getElementById("event-text-date").textContent = date;
+		document.getElementById("event-text-location").textContent = location;
+		var descStr = "";
+		for (line in description) {
```

Comment:
I think this should be ```for (var line in description) {``` to avoid the creation of a 'line' as a global variable. (```var``` will make a function-scoped variable, ```let``` will make a block-scoped variable (ie only exists in the for loop)).

Reply:
I'd prefer the use of `let` if possible, so as to be consistent with the rest of the codebase.

Reply:
changed to `for (let line in description) {`

Reply:
Yea I put var because I saw var being used at the top of the function, I agreee let is the best